lib/puppet/provider/package/homebrew.rb

```diff
@@ -8,18 +8,20 @@
   def install
     name = install_name
 
-    Puppet.debug "Installing #{name}"
-    output = execute([command(:brew), :install, name, *install_options])
-
-    if output =~ /Searching taps/
-      Puppet.debug "Falling back to brew-cask (still installing #{name}"
-      output = execute([command(:brew), :cask, :install, name, *install_options])
-      # brewcask includes some funky beer characters that f*ck with encoding
-      output = output.encode('UTF-8', :invalid => :replace, :undef => :replace)
-
+    begin
+      Puppet.debug "Looking for #{name} package on brew..."
+      output = execute([command(:brew), :info, name])
       if output.empty?
```

Comment:
Did try with a nested rescue but doesn't work. I can make it continue but after executing everything raises the error anyway. Any ideas how can be done without parsing?

```
 def install
    name = install_name

    begin
      Puppet.debug "Looking for #{name} package on brew..."
      output = execute([command(:brew), :info, name], failonfail: true)
      Puppet.debug "Package found, installing..."
      output = execute([command(:brew), :install, name, *install_options], failonfail: true)
      if output =~ /sha256 checksum/
        Puppet.debug "Fixing checksum error..."
        mismatched = output.match(/Already downloaded: (.*)/).captures
        fix_checksum(mismatched)
      end
    rescue => detail_brew
      next
      begin
        Puppet.debug "Package #{name} not found on Brew. Trying BrewCask..."
        output = execute([command(:brew), :cask, :info, name], failonfail: true)
        Puppet.debug "Package found on brewcask, installing..."
        output = execute([command(:brew), :cask, :install, name, *install_options], failonfail: true)
      rescue => detail_cask
        raise Puppet::Error, "Could not install package: #{detail_cask}"
      end
      raise Puppet::Error, "Could not install package: #{detail_brew}"
    end
  end
```